apps/hash-api/src/graph/system-types.ts

```diff
@@ -143,7 +146,8 @@ export let SYSTEM_TYPES: {
     usesUserSecret: EntityTypeWithMetadata;
 
     // Mention Notification related
-    occurredInEntity: EntityTypeWithMetadata;
+    occurredInPage: EntityTypeWithMetadata;
```

Comment:
Because we already have `occurredInText`/`occurredInComment`/`occurredInBlock`, I found `occurredInEntity` a bit too broad to distinguish from those other links.

What would you think about using `occurredInBlockCollection` instead?

Otherwise if you want the link to be more generic and keep `occurredInEntity` then maybe we should consider getting rid of `occurredInText`/`occurredInComment`/`occurredInBlock` and just check the destination type of the link to determine what is being referred to as you suggest, but this has the shortcoming that we aren't able to enforce that the notification has one of each in the source entity type.

Reply:
We might want people to be able to comment on other sorts of pages, e.g. an entity's page (which isn't a block collection, e.g. a single entity). 

`occurredInText`/`occurredInComment`/`occurredInBlock` are all additional specific information about _where_ in a page/entity the notification occurred. The last one dictates which route we need to send the user to (a page, a note, some other entity's page). Because we might have other routes, I think it's probably better to keep it broad. We could always call it something like `occurredInPageOfEntity`, and then we know what the route is based on what type of entity it is?

Reply:
We might conceivably want `occurredInProperty` at some point, if people are commenting on the properties of a single entity

Reply:
Have renamed it to `occurredInEntity` again in https://github.com/hashintel/hash/pull/3511/commits/b0872ef6776b692af3aa40929ec42673fc6ce727, I still think it'd be better if we could more easily distinguish this link from `occurredInText`/`occurredInComment`/`occurredInBlock` somehow but haven't come up with something better.